### PR TITLE
Feature/cfos 102

### PR DESCRIPTION
**Pull request by afonsobspinto**
Alert Deployment:
Upon merge it requires a re-ingestion of the bucket data

closes https://metacell.atlassian.net/browse/CFOS-102

- Creates viewerProbe component that is responsible to create a new amiJs probe widget anytime the scene changes (it relies on the probeVersion state variable of the viewer, which is updated on atlas, activityMap or slice changes)
- Adds baseAmiJs widget which is a 1-to-1 copy of [amiJS's widgets.base](https://github.com/FNNDSC/ami/blob/9e97472a0d24ee9dcab3d8cf8881351d6630a1ff/src/widgets/widgets.base.ts) (this was necessary because they do not expose that widget)
- Adds probeWidget which is based on [amiJS's widgets.voxelProbe](https://github.com/FNNDSC/ami/blob/9e97472a0d24ee9dcab3d8cf8881351d6630a1ff/src/widgets/widgets.voxelProbe.js). It replaces the dom manipulation with a callback + some other minor changes on behaviour. I believe this widget can be trimmed further but is not very clear so I propose we address that as technical debt.
- Updates deserialize function to properly handle the Ami js ModelFrame
- Adds LUT json to public folder (created from the csv provided by the customer but transformed into a format that facilitates lookups)
- Updates redux FETCH_MODEL action to add the LUT to the redux state.
- Connects tooltip and mapDetails component.

https://github.com/MetaCell/cfos-visualizer/assets/19196034/62a0ca1f-a3a3-4f44-a33e-60a8ffa30255

https://github.com/MetaCell/cfos-visualizer/assets/19196034/84e29705-e6a5-46fd-be83-1f3154501e11

closes https://metacell.atlassian.net/browse/CFOS-92

- Adds resizeObserverRef to observe and react to changes on the viewer container element.

Misc:

- Removes non-used webpack config files
- Adds mock_download_as_stream to server (to speed up development). Connected by default.

To do:

- [x] Update processor to include nifiti datatypeCode (Done here: https://github.com/MetaCell/cfos-visualizer/pull/40)
- [x] Address atlas wireframe
- [x] Position tooltip at the mouse location (done by @Salam-Dalloul thanks 🙏 )
- [x] Some UI look and feel touches (like not showing undefined f.e., show only visible activity maps, show only abbreviation in the tooltip, fix change slices onWheel)
- [x] Make brain use more of the canvas (props to @ddelpiano )
- [x] Fix slice out of range (coming from https://github.com/MetaCell/cfos-visualizer/pull/30/files)
- [x] Fix controls (props to @ddelpiano )
- [x] Fix gradient, in the image below we have a binary intensity so there should be no gradient when 0 is not included (new card created https://metacell.atlassian.net/browse/CFOS-127)

Before:
image

Now:
image

### CODE REVIEW

**Comment by afonsobspinto** [comment]
Given the map details introduction I don't see much value in the tooltip anymore. I find it a bit distracting and I don't like that it  overlaps with the viewer

**Review by aranega — CHANGES_REQUESTED** [review]
All looks good :) there is perhaps just a question I have about one line and a syntax I'm not aware of, beside that all nice! :+1:

File: client/src/components/Viewer.js [hunk 1]
```diff
@@ -123,15 +153,16 @@ export const Viewer = (props) => {
...
+        if (currentAtlas && sliceIndex && sliceIndex > 0) {
+            updateSliceIndex(currentAtlas, DIRECTIONS.DOWN, DELTA_SLICE_BUTTON);
         }
     };
 
     const handleNextSlice = () => {
         const currentAtlas = currentAtlasStackHelperRef.current;
-        if (sliceIndex && currentAtlas && sliceIndex < currentAtlas.orientationMaxIndex - 1) {
-            handleScrollHelper(currentAtlasStackHelperRef.current, DIRECTIONS.UP, DELTA_SLICE_BUTTON);
+        if (currentAtlas, sliceIndex && sliceIndex < currentAtlas.orientationMaxIndex - 1) {
```
[the remark this answers is not in the document]
Comment (afonsobspinto) [reply client/src/components/Viewer.js]: It's a typo. Thanks. Great eye 🦅

**Review by aranega — APPROVED** [review]
Thanks! I really wasn't sure it was a hidden js syntax feature I didn't know :sweat_smile:

**Comment by aranega** [comment]
I'll use your branch as base for CFOS-71

**Comment by ddelpiano** [comment]
@afonsobspinto could you please solve the conflicts here? Thanks!

**Comment by afonsobspinto** [comment]
> @afonsobspinto could you please solve the conflicts here? Thanks!

done ✅

**Review by aranega — APPROVED** [review]

**Pull request merged**